public/app/features/explore/TraceView.tsx

```diff
@@ -0,0 +1,247 @@
+      if (s.depth <= prevExpandedDepth) {
+        expandNextHiddenSpan = true;
+      }
+      if (expandNextHiddenSpan && res.has(s.spanID)) {
+        res.delete(s.spanID);
+        expandNextHiddenSpan = false;
+        prevExpandedDepth = s.depth;
+      }
+      return res;
+    }, new Set(childrenHiddenIDs));
+    setChildrenHiddenIDs(newChildrenHiddenIDs);
+  }
+
+  function collapseOne(spans: Span[]) {
+    if (shouldDisableCollapse(spans, childrenHiddenIDs)) {
+      return;
+    }
+    let nearestCollapsedAncestor: Span | undefined;
+    const newChildrenHiddenIDs = spans.reduce((res, curSpan) => {
+      if (nearestCollapsedAncestor && curSpan.depth <= nearestCollapsedAncestor.depth) {
+        res.add(nearestCollapsedAncestor.spanID);
+        if (curSpan.hasChildren) {
+          nearestCollapsedAncestor = curSpan;
+        }
+      } else if (curSpan.hasChildren && !res.has(curSpan.spanID)) {
+        nearestCollapsedAncestor = curSpan;
+      }
+      return res;
+    }, new Set(childrenHiddenIDs));
+    // The last one
+    if (nearestCollapsedAncestor) {
+      newChildrenHiddenIDs.add(nearestCollapsedAncestor.spanID);
+    }
+    setChildrenHiddenIDs(newChildrenHiddenIDs);
+  }
+
+  function expandAll() {
+    setChildrenHiddenIDs(new Set<string>());
+  }
+
+  function collapseAll(spans: Span[]) {
+    if (shouldDisableCollapse(spans, childrenHiddenIDs)) {
+      return;
+    }
+    const newChildrenHiddenIDs = spans.reduce((res, s) => {
+      if (s.hasChildren) {
+        res.add(s.spanID);
+      }
+      return res;
+    }, new Set<string>());
+
+    setChildrenHiddenIDs(newChildrenHiddenIDs);
+  }
+
+  function childrenToggle(spanID: string) {
+    const newChildrenHiddenIDs = new Set(childrenHiddenIDs);
+    if (childrenHiddenIDs.has(spanID)) {
+      newChildrenHiddenIDs.delete(spanID);
+    } else {
+      newChildrenHiddenIDs.add(spanID);
+    }
+    setChildrenHiddenIDs(newChildrenHiddenIDs);
+  }
+
+  function detailLogItemToggle(spanID: string, log: Log) {
+    const old = detailStates.get(spanID);
+    if (!old) {
+      return;
+    }
+    const detailState = old.toggleLogItem(log);
+    const newDetailStates = new Map(detailStates);
+    newDetailStates.set(spanID, detailState);
+    return setDetailStates(newDetailStates);
+  }
+
+  function addHoverIndentGuideId(spanID: string) {
+    setHoverIndentGuideIds(prevState => {
+      const newHoverIndentGuideIds = new Set(prevState);
+      newHoverIndentGuideIds.add(spanID);
+      return newHoverIndentGuideIds;
+    });
+  }
+
+  function removeHoverIndentGuideId(spanID: string) {
+    setHoverIndentGuideIds(prevState => {
+      const newHoverIndentGuideIds = new Set(prevState);
+      newHoverIndentGuideIds.delete(spanID);
+      return newHoverIndentGuideIds;
+    });
+  }
+
+  const traceProp = transformTraceData(props.trace);
+
+  return (
+    <UIElementsContext.Provider
+      value={{
+        Popover: (() => null as any) as any,
```

Comment:
Will need to be implemented, right now this means some of the functionality does not work, some buttons are missing or UI does not look as it should.

Reply:
Hi, look like this reason for my case #35939